packages/@aws-cdk/aws-kinesis/lib/stream.ts

```diff
@@ -290,10 +291,26 @@ export class Stream extends StreamBase {
    * user's configuration.
    */
   private parseEncryption(props: StreamProps): {
-    streamEncryption?: CfnStream.StreamEncryptionProperty,
+    streamEncryption?: CfnStream.StreamEncryptionProperty | IResolvable
     encryptionKey?: kms.IKey
   } {
 
+    // if encryption properties are not set, default to KMS in regions where KMS is available
+    if (!props.encryption && !props.encryptionKey) {
+      const condition = new CfnCondition(this, 'Condition', {
```

Comment:
Just thought about this... Maybe make this Condition a singleton in a Stack? In case there's more than one `Stream` instance used in the Stack?

Reply:
> In case there's more than one Stream instance used in the Stack?

Would be nice to uni-test that behavior as well :)

Reply:
that's a really good callout. I think it makes sense to avoid creating multiple conditions when they provide the same value. will try my hand at using it!

Reply:
added